axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/AbstractFixedAssetLineServiceImpl.java

```diff
@@ -237,6 +256,19 @@ public void filterListByStatus(List<FixedAssetLine> fixedAssetLineList, int stat
     clear(linesToRemove);
   }
 
+  @Override
+  public void filterListByDate(List<FixedAssetLine> fixedAssetLineList, LocalDate date) {
+
+    List<FixedAssetLine> linesToRemove = new ArrayList<>();
+    if (fixedAssetLineList != null) {
+      fixedAssetLineList.stream()
+          .filter(line -> line.getDepreciationDate().isAfter(date))
+          .forEach(line -> linesToRemove.add(line));
+      fixedAssetLineList.removeIf(line -> line.getDepreciationDate().isAfter(date));
+    }
+    clear(linesToRemove);
```

Comment:
I think it would be better with a single for loop. If condition is true, the element is removed from the list and deleted with the method named `remove`